src/control/destination/background/client.rs

```diff
@@ -46,54 +42,24 @@ pub(super) struct BindClient {
     log_ctx: ::logging::Client<&'static str, HostAndPort>,
 }
 
-/// Wait a duration if inner `poll_ready` returns an error.
-//TODO: move to tower-backoff
-struct Backoff<S> {
-    inner: S,
-    timer: Delay,
-    waiting: bool,
-    wait_dur: Duration,
-}
-
-/// Log errors talking to the controller in human format.
-struct LogErrors<S> {
-    inner: S,
-}
-
-// We want some friendly logs, but the stack of services don't have fmt::Display
-// errors, so we have to build that ourselves. For now, this hard codes the
-// expected error stack, and so any new middleware added will need to adjust this.
-//
-// The dead_code allowance is because rustc is being stupid and doesn't see it
-// is used down below.
-#[allow(dead_code)]
-type LogError = ReconnectError<
-    tower_h2::client::Error,
-    tower_h2::client::ConnectError<
-        TimeoutError<
-            io::Error
-        >
-    >
->;
-
 // ===== impl ClientService =====
 
 impl svc::Service for ClientService {
     type Request = http::Request<BoxBody>;
     type Response = http::Response<RecvBody>;
-    type Error = LogError;
-    type Future = ReconnectFuture<
-        tower_h2::client::Connect<
-            Timeout<LookupAddressAndConnect>,
-            ::logging::ContextualExecutor<
-                ::logging::Client<
-                    &'static str,
-                    HostAndPort,
-                >
-            >, BoxBody>>;
+    type Error = <Service as svc::Service>::Error;
+    type Future = <Service as svc::Service>::Future;
 
     fn poll_ready(&mut self) -> Poll<(), Self::Error> {
-        self.0.poll_ready()
+        // If an error occurs on established connection, log it; otherwise the `reconnect` module will handle re-establishing a client.
+        loop {
```

Comment:
For a minute I thought this might be better expressed with a `while let` loop, but realized that wouldn't bind the `Ok` case as easily, so nevermind.